pkg/infrastructure/azure/azure.go

```diff
@@ -0,0 +1,356 @@
+		Region:             platform.Region,
+		Tags:               tags,
+		TokenCredential:    tokenCredential,
+		CloudConfiguration: cloudConfiguration,
+	})
+	if err != nil {
+		return err
+	}
+
+	computeClientFactory := createImageGalleryOutput.ComputeClientFactory
+
+	// Create gallery images
+	_, err = CreateGalleryImage(ctx, &CreateGalleryImageInput{
+		ResourceGroupName:    resourceGroupName,
+		GalleryName:          galleryName,
+		GalleryImageName:     galleryImageName,
+		Region:               platform.Region,
+		Tags:                 tags,
+		TokenCredential:      tokenCredential,
+		CloudConfiguration:   cloudConfiguration,
+		OSType:               armcompute.OperatingSystemTypesLinux,
+		OSState:              armcompute.OperatingSystemStateTypesGeneralized,
+		HyperVGeneration:     armcompute.HyperVGenerationV1,
+		Publisher:            "RedHat",
+		Offer:                "rhcos",
+		SKU:                  "basic",
+		ComputeClientFactory: computeClientFactory,
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = CreateGalleryImage(ctx, &CreateGalleryImageInput{
+		ResourceGroupName:    resourceGroupName,
+		GalleryName:          galleryName,
+		GalleryImageName:     galleryGen2ImageName,
+		Region:               platform.Region,
+		Tags:                 tags,
+		TokenCredential:      tokenCredential,
+		CloudConfiguration:   cloudConfiguration,
+		OSType:               armcompute.OperatingSystemTypesLinux,
+		OSState:              armcompute.OperatingSystemStateTypesGeneralized,
+		HyperVGeneration:     armcompute.HyperVGenerationV1,
+		Publisher:            "RedHat-gen2",
+		Offer:                "rhcos-gen2",
+		SKU:                  "gen2",
+		ComputeClientFactory: computeClientFactory,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create gallery image versions
+	_, err = CreateGalleryImageVersion(ctx, &CreateGalleryImageVersionInput{
+		ResourceGroupName:       resourceGroupName,
+		StorageAccountID:        *storageAccount.ID,
+		GalleryName:             galleryName,
+		GalleryImageName:        galleryImageName,
+		GalleryImageVersionName: galleryImageVersionName,
+		Region:                  platform.Region,
+		BlobURL:                 blobURL,
+		RegionalReplicaCount:    int32(1),
+		ComputeClientFactory:    computeClientFactory,
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = CreateGalleryImageVersion(ctx, &CreateGalleryImageVersionInput{
+		ResourceGroupName:       resourceGroupName,
+		StorageAccountID:        *storageAccount.ID,
+		GalleryName:             galleryName,
+		GalleryImageName:        galleryGen2ImageName,
+		GalleryImageVersionName: galleryGen2ImageVersionName,
+		Region:                  platform.Region,
+		BlobURL:                 blobURL,
+		RegionalReplicaCount:    int32(1),
+		ComputeClientFactory:    computeClientFactory,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func randomStringPrefixFunction() string {
+	length := 5
+	rand.Seed(time.Now().UnixNano())
```

Comment:
nit: this rand.Seed is deprecated

Reply:
Copied and pasted from elsewhere in the installer. Will update ;-)

Reply:
Fixed.